src/java/org/jivesoftware/LoginDialog.java

```diff
@@ -277,11 +239,22 @@ protected XMPPTCPConnectionConfiguration retrieveConnectionConfiguration() {
         }
 
         final XMPPTCPConnectionConfiguration.Builder builder = XMPPTCPConnectionConfiguration.builder()
-                .setUsernameAndPassword( "username", "password" )
+                .setUsernameAndPassword( loginUsername, loginPassword )
                 .setServiceName( loginServer )
                 .setPort( port )
                 .setSendPresence( false )
                 .setCompressionEnabled( localPref.isCompressionEnabled() );
+        try {
+            TLSUtils.acceptAllCertificates(builder);
```

Comment:
Lets make this configurable. Please add an `isAcceptAllCertificates()` and `setAcceptAllCertificates(boolean accept)` and make it default to 'true'. You can then add an "if" condition to this line.

For brownie points, add the setting to `LoginSettingDialog`. :)